### PR TITLE
Components: update components provider story

### PR DESCRIPTION
**Pull request by ciampo**
## Description

Following #32566, this PR cleans up the `ComponentsProvider` storybook story

This PR was mainly created to highlight a bug in the g2 Context system, which was fixed by #32745.

## Testing instructions

- [ ] Project compiles, all tests pass
- [ ] Check out the new story. On your machine, run `npm run storybook:dev` and visit [`http://localhost:50240/?path=/story/g2-components-experimental-contextsystemprovider--default`](http://localhost:50240/?path=/story/g2-components-experimental-contextsystemprovider--default) 
    - [ ] Check the Story's code, and make sure that it behaves like expected, especially around the Context (i.e. the bugs reported [below](https://github.com/WordPress/gutenberg/pull/32743#issuecomment-862552895) should all be fixed)

## Screenshots 

Screenshot 2021-06-22 at 11 58 20

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by ciampo** [comment]
Working on this refactor has exposed some unexpected behaviour from the Context system:

## Values from contexts leak to components outside of the context provider

image

On top of this, if this patch is applied, the resulting behaviour seems also unexpected:

```diff
diff --git a/packages/components/src/ui/context/stories/ComponentsProvider.stories.js b/packages/components/src/ui/context/stories/ComponentsProvider.stories.js
index 75f6414129..bed36eff14 100644
--- a/packages/components/src/ui/context/stories/ComponentsProvider.stories.js
+++ b/packages/components/src/ui/context/stories/ComponentsProvider.stories.js
@@ -31,6 +31,11 @@ const outerContext = {
                        display: 'block',
                },
        },
+       Text: {
+               style: {
+                       outline: '2px solid orange',
+               },
+       },
 };
 
 const innerContext = {
```

image


## Incrementing the counter (force update) makes the inner card component lose the `elevation` value from the context 

https://user-images.githubusercontent.com/1083581/122261331-9437b700-ced4-11eb-8fe1-e231dddaec57.mp4

Note that this happens only when `as="div"` is applied on the inner card via context (instead of applying the prop directly on the inner `CardBody`)

## When using new version of `<Card />`, context values are not applied at all

This can be reproduced by applying this diff:

```diff
diff --git a/packages/components/src/ui/context/stories/ComponentsProvider.stories.js b/packages/components/src/ui/context/stories/ComponentsProvider.stories.js
index 75f6414129..181ef57935 100644
--- a/packages/components/src/ui/context/stories/ComponentsProvider.stories.js
+++ b/packages/components/src/ui/context/stories/ComponentsProvider.stories.js
@@ -6,7 +6,7 @@ import { createContext, useContext, memo, useState } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import { Card, CardBody } from '../../';
+import { Card, CardBody } from '../../../card';
 import { View } from '../../../view';
 import { Text } from '../../../text';
 import { ContextSystemProvider } from '../index';
```

This probably has to do with the fact that the new version of `<Card />` internally also [uses the context system for the same keys](https://github.com/WordPress/gutenberg/blob/707d2654ca9c03e11a06d99bb1643ef102772d8f/packages/components/src/card/card/component.js#L41-L51)

image


## Overall unreliability of what's rendered in storybook when fast-refreshing (in memory)

I also noticed that often, when making changes to this story in particular, what is rendered in the browser after a fast refresh is often different from what is rendered after a full, hard refresh (e.g. sometimes the bugs above don't show when fast refreshing)

**Comment by ciampo** [comment]
Rebased on top of `trunk` in order to include the fix from #32745 

@sarayourfriend could you have a look at this?

File: packages/components/src/ui/context/stories/ComponentsProvider.stories.js [hunk 1]
```diff
@@ -19,20 +19,39 @@ export default {
...
+			display: 'block',
+		},
+	},
+};
+
 const innerContext = {
 	Card: {
-		css: {
+		style: {
 			background: 'white',
```
Comment (sarayourfriend) [inline packages/components/src/ui/context/stories/ComponentsProvider.stories.js]: Should this be something other than white, something we can actually see the difference of?

**Review by sarayourfriend — APPROVED** [review]
LGTM! Thanks!

File: packages/components/src/ui/context/stories/ComponentsProvider.stories.js (hunk 1 above)
Comment (ciampo) [reply packages/components/src/ui/context/stories/ComponentsProvider.stories.js]: Good point, I changed it to a shade of green in https://github.com/WordPress/gutenberg/pull/32743/commits/aff830ca84ca7b21bb9d1f4e781332961c3cf07b

**Pull request merged**